pkg/platform/packet/packet.go

```diff
@@ -237,6 +255,100 @@ func (c *config) terraformSmartApply(ex *terraform.Executor, dnsProvider dns.DNS
 	return ex.Apply()
 }
 
+// terraformAddDependencies adds explicit dependencies to cluster nodes so nodes
+// with a specific hw reservation UUID are created before nodes that don't have
+// a specific hw reservation UUID.
+// The function modifies c.NodesDependsOn and c.workerPools[].NodesDependsOn,
+// assigning to them a string, formatted as a terraform list, of all worker
+// terraform targets needed to be created first. For example:
+//
+//	// Suppose worker pool "example" is the only to use a specific hw
+//	// reservation IDs. IOW, the controllers (c.NodesDependsOn) depends on
+//	// worker pool "example" to be created first.
+//	// Then, after calling this function, the attribute will be:
+// 	c.NodesDependsOn = "[ module.worker-example.worker_nodes_ids ]"
+//
+//
+// The explicit terraform dependency is needed to guarantees that nodes using
+// hardware reservation "next-available" won't use reservation IDS that another
+// worker pool may specify with a specific UUID, and thus fail to create the
+// node. This race condition is best explained here, if you want more info:
+// https://github.com/terraform-providers/terraform-provider-packet/issues/176
+// https://github.com/terraform-providers/terraform-provider-packet/pull/208
+func (c *config) terraformAddDependencies() {
+	// Nodes with specific hw reservation IDs.
+	nodesWithRes := make([]string, 0)
+
+	// Note that dependencies expressed in terraform are using the module
+	// ouput "controller_nodes_ids" for controllers and "worker_nodes_ids"
```

Comment:
Hm, could we unify that and just have output parameter `device_ids` ?

Reply:
I went that way because workers already use the `worker_` prefix for its outputs (`worker_nodes_hostname` and friends), for consistency.

IMHO, it's the same. The string is shorter but clear (that is good), but won't allow to remove if branches anyways, so... I'm okay either way.

If we are going to use the same, though, I think I prefer `nodes_ids`, as it is even shorter, is a little bit clearer it refers to server instances and it's more consistent with `nodes_depends_on` naming. What do you think?

Reply:
I choose `device_ids`, because Packet resource is called `packet_device` and has `id` attribute, so `device_ids` seem natural. `node_ids` would also be okay I guess. I don't really like `nodes_ids` personally.

Reply:
Okay, switched to `device_ids`.

Just curious, what is it that you don't like about the plural form (devices_ids or nodes_ids)?